### ISSUE TITLE
Redirect to signup

### ISSUE THREAD

**Issue by mbenitezm** [auth0/omniauth-auth0#87]
Hi!

I'm on a situation where I have both a login and signup in my home page. I cannot find a way to redirect to auth0 with signup page as the initial screen, it will always show login instead.

I see javascript have some options in which you can set the initial screen, but I cannot find the equivalent to this using this gem.

Is there any way I can redirect my users to the signup screen when they click signup?

Thanks a lot!

**Comment by neilmock** [comment]
I am having the same issue, any help is welcome!  👍

**Comment by joshcanhelp** [comment]
Apologies for the long delay. You should be able to add `prompt=signup` to your login link and that will pass along to the login page at Auth0. If you're having trouble there, ping this thread and we'll take a look!

**Comment by mbenitezm** [comment]
Hi Josh, thanks for your help!

I'm doing as you said and it's not working for me.

`'/auth/auth0?prompt=signup'`

This is the link I'm using when redirecting to auth0 from my site. Am I not understanding something here?

Again, thanks a lot for your help!

**Comment by joshcanhelp** [comment]
Happy to help! If you follow along in your browser, does that parameter get sent to Auth0 in the redirect?

**Comment by mbenitezm** [comment]
Yes, it is even sent to Google. That makes google-auth login to fail
because it receives an invalid prompt.

Thanks for the fast reply!

**Comment by neilmock** [comment]
No luck here either, regardless of any querystring param on /auth/auth0 I'm always directed to

https://auth.rocketphoto.com/u/login?state=<token>

**Comment by joshcanhelp** [comment]
@mbenitezm - I'm a little confused ... are you trying to force signup on Auth0? How does Google come into play here?

@neilmock - I'm confused on yours as well. You're saying that `state` is being added to the login URL?

**Comment by mbenitezm** [comment]
So I have a button that redirects to auth0 login, just as shown in the auth0 rails tutorial.

The url of that button is `'/auth/auth0?prompt=signup'`

When redirected to the auth0 subdomain I do not see the signup form even after adding that prompt.

Once that I'm on the login form, ignoring that I didn't see the signup form, when I click login with Google and get redirected to google consent form, Google shows me an error saying that I sent an invalid prompt in the url.

So in summary, I don't see the signup form when adding the params that you mentioned. And also this param gets sent all the way to google after clicking google login on auth0 subdomain,

**Comment by joshcanhelp** [comment]
@mbenitezm - Thank you for the clarification here, happy to look into both of these. One more question for you ... are you using the Classic or the New universal login form?

**Comment by mbenitezm** [comment]
I'm using new login.

Again, thanks for your help!

**Comment by tjbarber** [comment]
Just nudging this to see if there's any solution yet.

I have a Rails app and would like to open the sign up form when clicking a specific link and the login form when clicking another one. Right now I can only see the log in form. Is there a way to have Auth0 open the with sign up form?

**Comment by neilmock** [comment]
> Just nudging this to see if there's any solution yet.
> 
> I have a Rails app and would like to open the sign up form when clicking a specific link and the login form when clicking another one. Right now I can only see the log in form. Is there a way to have Auth0 open the with sign up form?

I gave up and used the Lock JS library directly, which has all the configuration params available.

**Comment by mbenitezm** [comment]
Using Lock JS is not currently an option for me, so I'm still hoping this gets solved.

**Comment by joshcanhelp** [comment]
Apologies for the delay here, folks. Hopefully this is something we can address easily in the library or with a config change but we have to dig in and really figure out where the problem lies. 

We definitely recommend sticking with redirect-based login rather than the embedded form.

**Comment by shafy** [comment]
Any updates here? When I add query params to `auth/auth0´, (such as `auth/auth0?prompt=signup´), it doesn't even persist them when redirecting to my Universal Login Page. I just see when query params which is `state=xyz`. Is this a bug or intended?
